integration_test/cases/info_test.exs

```diff
@@ -0,0 +1,60 @@
+defmodule InfoTest do
+  use ExUnit.Case, async: true
+  alias TestPool, as: P
+  alias TestAgent, as: A
+
+  test "handle_info handles message and moves on" do
+    stack = [
+      fn opts ->
+        send(opts[:parent], {:connected, self()})
+        {:ok, :state}
+      end,
+      :ok,
+      {:idle, :state},
+      {:idle, :state}
+    ]
+
+    {:ok, agent} = A.start_link(stack)
+    {:ok, pool} = P.start_link(agent: agent, parent: self())
+
+    assert_receive {:connected, conn}
+    send(conn, "some harmless message")
+    assert P.run(pool, fn _ -> :result end) == :result
```

Comment:
I'm using this just to "sync" with the connection process, i.e. wait once it processed all the messages. Maybe there is a better way? I usually use `:sys.get_state/1` but don't know who well it would work with DBConnection.